### PR TITLE
[Scalar Function][Array] Add ArrayJoinString Scalar function

### PR DESCRIPTION
**Pull request by nikhilsu**
Adding ArrayJoinString Scalar function that performs a String.Join on an array of string values.

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java [hunk 1]
```diff
@@ -347,4 +347,9 @@ public static double[] generateDoubleArray(double start, double end, double inc)
     }
     return arr;
   }
+
+  @ScalarFunction
+  public static String arrayJoinString(String[] values, String delimiter) {
```
Comment (mayankshriv) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:352]: Is there a Postgres equivalent? Does it make sense to use the same/similar name?

Comment (xiangfu0) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:352]: Postgres has a function called: `ARRAY_TO_STRING` : https://www.postgresql.org/docs/9.1/functions-array.html

I think we can implement this signature.

Comment (xiangfu0) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:352]: Can you change the function name to `arrayToString`

Comment (xiangfu0) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:352]: For null handling you need to create a new function:
```
public static String arrayToString(String[] values, String delimiter, String nullString) {
...
```

Comment (nikhilsu) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:352]: @xiangfu0 done. PTAL - need your eyes on the null checker.

**Review by xiangfu0 — APPROVED** [review]

**Comment by xiangfu0** [comment]
Thanks for supporting this !

**Pull request merged**